src/main/java/com/google/cloud/tools/libraries/CloudLibraries.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.libraries;
+
+import com.google.cloud.tools.libraries.json.CloudLibrary;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.util.List;
+
+/** Returns helpful metadata for supported Google Cloud libraries. */
+public final class CloudLibraries {
```

Comment:
This is going to have a name conflict with existing classes in CT4E.

Reply:
Is the package the same? If the package isn't the same, then I don't believe there's an issue. If CT4E migrates to use these classes, I doubt it would reference both at the same time, requiring it to fully specify the class references.

Reply:
The package isn't the same, but it's still inconvenient to have classes with the same name, if not exactly impossible.

Reply:
I think that if we can agree to share the parsing logic across clients in appengine-plugins-core then this naming issue will go away.